### DISCUSSION TITLE
Non ad support

### DISCUSSION THREAD

**Discussion by uaygunn** [gravitational/teleport#20766] [Q&A] [answered]
Hello guys i have some questions. 

Does v12 will have windows desktop access without active directory ? Because i downloaded alpha v12 and it still says it will be developing.

and second question if it will support non ad windows access, how will it be ? will it be with agent or will it create a rdp file from my credentials ?

Thank you for you answers.

**Comment by webvictim** [comment, marked as answer]
Teleport 12 will indeed support desktop access to non-AD Windows systems.

There will an agent install required on the Windows systems. The docs for the setup process are still in the process of being written and will be published when Teleport 12 is released.

**Reply by uaygunn** [reply]
Is not it easy to make it just create for say an rdp field (idk but if you used cyberark it does that thing). I am just curious about it otherwise of course agent is more secure way. But let's say i have 300 servers and all of them outside of active directory so i need to setup 300 agents on the machines which is hard thing. But on the otherhand if teleport can create custom rdp with my given credentials it would be life saving. I know that everyone will not using 300 machines but this comes to my mind.

**Reply by webvictim** [reply]
Teleport isn't a password broker, it uses short-lived certificates exclusively for communication between users and services. As such, an agent installation is mandatory. It's definitely a little more work to set up, but it's orders of magnitude more secure than storing a username and password which could be obtained and used by anyone.

The fact that agent installation and setup is difficult is part of the reason we went with an Active Directory based approach in the first place, where trust can be centralised rather than needing to be offloaded to individual machines.

**Reply by uaygunn** [reply]
I understood your point very clearly, for me it is also safe way is to use an agent based installation method. I was just wondering tho.
Thank you again for your clear answers :)
